Sources/Swift/Helper/DataDeserialization.swift

```diff
@@ -29,4 +31,154 @@
       }
     }
+    
+    //swiftlint:disable cyclomatic_complexity function_body_length
+    @objc(envelopeWithData:) public static func envelope(with data: Data) -> SentryEnvelope? {
+        var envelopeHeader: SentryEnvelopeHeader?
+        let bytes = [UInt8](data)
+        var envelopeHeaderIndex: Int = 0
+        
+        for i in 0..<data.count {
+            if bytes[i] == UInt8(ascii: "\n") {
+                envelopeHeaderIndex = i
+                // Envelope header end
+                let headerData = data.subdata(in: 0..<i)
+                #if DEBUG
+                  if let headerString = String(data: headerData, encoding: .utf8) {
+                    SentrySDKLog.debug("Header \(headerString)")
+                  }
+                #endif
+                do {
+                    let headerDictionary = try JSONSerialization.jsonObject(with: headerData) as? [String: Any]
+                    var eventId: SentryId?
+                    if let eventIdAsString = headerDictionary?["event_id"] as? String {
+                        eventId = SentryId(uuidString: eventIdAsString)
+                    }
+                    
+                    var sdkInfo: SentrySdkInfo?
+                    if let sdkDict = headerDictionary?["sdk"] as? [String: Any] {
+                        sdkInfo = SentrySdkInfo(dict: sdkDict)
+                    }
+
+                    var traceContext: TraceContext?
+                    if let traceDict = headerDictionary?["trace"] as? [String: Any] {
+                        traceContext = TraceContext(dict: traceDict)
+                    }
+                    
+                    envelopeHeader = SentryEnvelopeHeader(id: eventId,
+                                                      sdkInfo: sdkInfo,
+                                                      traceContext: traceContext)
+
+                    if let sentAtStr = headerDictionary?["sent_at"] as? String {
+                        envelopeHeader?.sentAt = sentry_fromIso8601String(sentAtStr)
+                    }
+                    break
+                } catch {
+                    SentrySDKLog.error("Failed to parse envelope header \(error)")
+                    break
+                }
+            }
+        }
+        
+        guard let envelopeHeaderUnwrapped = envelopeHeader else {
+            SentrySDKLog.error("Invalid envelope. No header found.")
+            return nil
+        }
+
+        if envelopeHeaderIndex == 0 {
+            SentrySDKLog.error("EnvelopeHeader was parsed, its index is expected.")
+            return nil
+        }
+        
+        // Parse items
+        var itemHeaderStart = envelopeHeaderIndex + 1
+        var items: [SentryEnvelopeItem] = []
+        let endOfEnvelope = data.count - 1
+
+        var i = itemHeaderStart
+        while i <= endOfEnvelope {
+            defer {
+                i += 1
+            }
+            if bytes[i] == UInt8(ascii: "\n") || i == endOfEnvelope {
```

Comment:
`l`: We could define a variable or `UInt8(ascii: "\n")` at the top of this method, so we don't have it twice.